File: backend/internal/models/history.go
package models

import "time"

// QueryHistory 查询历史模型
type QueryHistory struct {
	ID              int       `json:"id" db:"id"`
	UserID          int       `json:"user_id" db:"user_id"`
	ConnectionID    int       `json:"connection_id" db:"connection_id"`
	SQLQuery        string    `json:"sql_query" db:"sql_query"`
	ExecutionTimeMs int       `json:"execution_time_ms" db:"execution_time_ms"`
	RowsAffected    int       `json:"rows_affected" db:"rows_affected"`
	Success         bool      `json:"success" db:"success"`
	ErrorMessage    string    `json:"error_message" db:"error_message"`
	CreatedAt       time.Time `json:"created_at" db:"created_at"`
}

// QueryHistoryResponse 查询历史响应
type QueryHistoryResponse struct {
	ID              int       `json:"id"`
	ConnectionID    int       `json:"connection_id"`
	ConnectionName  string    `json:"connection_name,omitempty"`
	SQLQuery        string    `json:"sql_query"`
	ExecutionTimeMs int       `json:"execution_time_ms"`
	RowsAffected    int       `json:"rows_affected"`
	Success         bool      `json:"success"`
	ErrorMessage    string    `json:"error_message"`
	CreatedAt       time.Time `json:"created_at"`
}

// QueryHistoryListResponse 查询历史列表响应
type QueryHistoryListResponse struct {
	List     []QueryHistoryResponse `json:"list"`
	Total    int                    `json:"total"`
	Page     int                    `json:"page"`
	PageSize int                    `json:"page_size"`
}

File: frontend/src/types/history.ts
/**
 * 历史记录类型定义
 */

/**
 * 查询历史记录
 */
export interface QueryHistory {
  id: number
  connection_id: number
  connection_name?: string
  sql_query: string
  operation_type?: string // 系统执行记录才有此字段
  execution_time_ms: number
  rows_affected: number
  success: boolean
  error_message?: string
  created_at: string
}

/**
 * 查询历史列表响应
 */
export interface QueryHistoryListResponse {
  list: QueryHistory[]
  total: number
  page: number
  page_size: number
}
